master/buildbot/status/buildstep.py

```diff
@@ -397,7 +405,7 @@ def asDict(self, request=None):
                 for l in self.getLogs()]
 
         result["urls"] = self.getURLs()
-
+        result["urls"].update(self.getArtifactURLs())
```

Comment:
I agree we should have 'artifacts' separated in json result, but we need to be careful as this will break the API. We need to have this fully tested with the system that uses json before releasing it.